compiler/src/dotty/tools/dotc/typer/TyperPhase.scala

```diff
@@ -103,19 +75,27 @@ class FrontEnd extends Phase {
   override def runOn(units: List[CompilationUnit])(using Context): List[CompilationUnit] =
     val unitContexts =
       for unit <- units yield
-        report.inform(s"compiling ${unit.source}")
-        ctx.fresh.setCompilationUnit(unit).withRootImports
-    unitContexts.foreach(parse(using _))
-    record("parsedTrees", ast.Trees.ntrees)
+        val newCtx = ctx.fresh.setPhase(this.start).setCompilationUnit(unit)
+        report.inform(s"typing ${unit.source}")
+        // TODO without this test, dotty.tools.repl.ScriptedTests fails. Not sure why.
+        if (this.start > Periods.FirstPhaseId)
```

Comment:
Rather than using this to indirectly check if we're in the REPL, I suggest defining `class TyperPhase(val addRootImports: Boolean = true)` and setting the parameter to false in ReplCompiler.

Reply:
Done